### DISCUSSION TITLE
On using for a plugin

### DISCUSSION THREAD

**Discussion by katawful** [Olical/nfnl#26] [Q&A] [answered]
Moving from [here](https://github.com/Olical/aniseed/discussions/148), I would like to use this for my macros repo for an easier time to write tests (compared to just writing from plenary direct), but I'm not sure how to do it exactly. The docs say it can be done, but from what I've read it seems like you need to have nfnl installed from the get go (which i don't wish to do atm until I've gotten to rewriting my configs)

**Comment by Olical** [comment, marked as answer]
You do need `nfnl` as a plugin yes, but it won't do anything unless you're working in a directory with an nfnl config file (`.nfnl.fnl`). So it's effectively doing nothing until you configure it for your project. You can then configure it to ONLY compile files within a certain directory, such as your tests directory. I think this would scope it, you would add this to `.nfnl.fnl`.

```fennel
{:source-file-patterns ["test/fnl/**/*.fnl"]}
```

Now when you write any changes to your test files they'll be compiled to `test/lua/...` but nothing else.

**Reply by Olical** [reply]
The other thing you'd need to do is mark your macro files with `;; [nfnl-macro]`, as shown in the readme https://github.com/Olical/nfnl#macros - this removes all ambiguity for the compiler. It doesn't use heuristics or names, you tell it which files are macros.

Every time you write to a macro file it'll then recompile all of the other non-macro files to ensure your Lua output stays in sync.

Although if you scope nfnl to just your test directory I don't think you'll get this, because you'll be making changes outside of the test dir when modifying your macro source files which will be ignored. Your source-file-patterns would probably need to include your macro file paths too for this to work.
